packages/core/src/core/coreToolScheduler.ts

```diff
@@ -819,7 +819,7 @@ export class CoreToolScheduler {
           );
         }
       }
-      this.attemptExecutionOfScheduledCalls(signal);
+      await this.attemptExecutionOfScheduledCalls(signal);
```

Comment:
is there a unit test for testing that tool calls are executed sequentially?

A recent PR added parallel tool calls. Is this just moving back to that version?